R/utils.R

```diff
@@ -92,3 +92,123 @@ browse_url <- function(path, chromote) {
     utils::browseURL(url)
   }
 }
+
+
+# =============================================================================
+# Random Ports
+# =============================================================================
+#
+# Borrowed from https://github.com/rstudio/httpuv/blob/main/R/random_port.R
+
```

Comment:
It would be very helpful if there was an explicit description of how `startup` is supposed to behave. It is hard to reason about the code below without knowing that information.

For example, something like:

* `startup` always returns a value if successful.
* If it times out, then it throws an error with class `"error_timeout"`
* If it fails for some other reason (like if the port is in use), then it throws an error.

Reply:
Also, I think the `startup` function does not need to take a `host` parameter -- it only needs `port`.